snow/engine/snowman/engine.go

```diff
@@ -875,6 +885,10 @@ func (e *Engine) sendQuery(
 		zap.Stringer("validators", e.Validators),
 	)
 
+	if e.abortDueToInsufficientConnectedStake(blkID) {
+		return
+	}
```

Comment:
Should we move this check to be done before logging `sampling from validators`? It feels weird to log that we are doing something and then potentially exiting _before_ doing that thing.

Reply:
yeah good point